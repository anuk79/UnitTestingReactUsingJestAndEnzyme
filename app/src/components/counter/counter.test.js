
// vendors
import React from "react";
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";

// components
import Counter from './counter';

describe("Counter component", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<Counter />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('div').length).toBe(1);
        expect(wrapper.find('p').text()).toBe('You clicked 0 times');
        expect(wrapper.find('button').length).toBe(1);
    });

    it("should change the count when button is clicked", () => {
        const wrapper = shallow(<Counter />);
        wrapper.find('button').simulate('click');
        expect(wrapper.find('p').text()).toBe('You clicked 1 times');
    });
});