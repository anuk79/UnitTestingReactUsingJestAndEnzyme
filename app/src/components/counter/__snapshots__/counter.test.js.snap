// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should render correctly 1`] = `
<div>
  <p>
    You clicked 
    0
     times
  </p>
  <button
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
